### PR TITLE
Reorganize Quickstart Guide

### PR DESCRIPTION
**Pull request by fitzthum**
We've let the quickstart guide get overgrown. This PR makes it much quicker.

I moved a lot of content into separate files. I don't want to de-emphasize anything, but putting everything in one file simply does not scale and it seriously compromises the focus of what is supposed to be an approachable starting point for people coming to the project. I have tried to include pointers to the new documents in every relevant section. I don't want any of the great work people have already done on the guide to be lost.

I ditched our old four-step structure (which we weren't really following anyway) and opted for a different four-step structure, which I think is a bit simpler. Now the quickstart guide focuses on installation, basic unencrypted workloads, encryption, singing, and attestation using the generic KBS. Everything else has been moved to other files.

Hopefully I'm not the only one who thinks this is an improvement because it was a huge pain. I think this should hold us over until we get serious about having a documentation website. lmk what you think.

### CODE REVIEW

File: quickstart.md [hunk 1]
```diff
@@ -222,184 +224,40 @@ Now go back to the k8s node and ensure that you still don’t have any bitnami/n
...
-
-Now you can use [simple-kbs](guides/sev-guide.md) to create encrypted container image and depolying it on SEV.
-Please refer to [sev-guide.md](guides/sev-guide.md)
+For `enclave-cc` follow the [enclave-cc guide](./guides/enclave-cc.md).
 
 ### Deploy and Configure tenant-side CoCo Key Broker System cluster
 
-A tenant-side CoCo Key Broker System cluster include:
+The follow describes how to run and provision the generic KBS.
+The KBS should be run in a trusted environment. Thee KBS is not just one service,
```
Comment (sameo) [inline quickstart.md:257]:
```suggestion
The KBS should be run in a trusted environment. The KBS is not just one service,
```

File: quickstart.md [hunk 2]
```diff
@@ -222,184 +224,40 @@ Now go back to the k8s node and ensure that you still don’t have any bitnami/n
...
+CoCo has a modular attestation interface and there are a few options for attestation.
+CoCo provides a generic Key Broker Service (KBS) that the rest of this guide will be focused on.
+The SEV runtime class uses `simple-kbs`, which is described in the [SEV guide](./guides/sev.md).
+There is also `eaa_kbc`/`verdictd` which is described [here](./guides/eaa_verdictd).
 
-We have prepared a sample CoCo operator custom resource that is based on the standard `ccruntime.yaml`, but in addition has the the decryption keys and configuration required to decrypt this sample container image.
-> **Note** All pods started with this sample resource will be able to decrypt the sample container and all keys shown are for demo purposes only and should not be used in production.
+To use CoCo with confidential hardware, first switch to the appropriate runtime class.
+TDX has two runtime classes, `kata-qemu-tdx` and `kata-clh-tdx`. One uses QEMU as VMM and TDVF as firmware. The other uses Cloud Hypervisor as VMM and TD-Shim as firmware.
```
Comment (sameo) [inline quickstart.md:248]: I would put that part under a `### Runtime Class` section.

File: quickstart.md [hunk 3]
```diff
@@ -189,6 +187,10 @@ spec:
   runtimeClassName: kata
 ```
 
+Setting the runtimeclass is usually the only change needed to the pod yaml, but some platforms
```
Comment (sameo) [inline quickstart.md:190]:
```suggestion
Setting the `runtimeClassName` is usually the only change needed to the pod yaml, but some platforms
```

**Review by sameo — COMMENTED** [review]
Very nice, thanks for putting this together. I only have a few nits.

File: quickstart.md [hunk 4]
```diff
@@ -1,28 +1,40 @@
...
+across a wide array of platforms and deployment models.
 
-Confidential Containers (CoCo) supports the following use cases:
+In general, using CoCo involves the following steps:
 
-- Running unencrypted containers without Confidential Computing (CC) hardware
-- Running encrypted containers without CC HW (sample container images provided)
-- Running unencrypted container images or sample encrypted images with CC HW
-- Running your own encrypted container images with CC HW
+- Install CoCo using the operator.
```
Comment (dcmiddle) [inline quickstart.md:8]: extraneous .
```suggestion
- Install CoCo using the operator
```

File: guides/enclave-cc.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
+# Enclave-CC
+
```
Comment (dcmiddle) [inline guides/enclave-cc.md:2]:
```suggestion
This guide assumes you have already completed **Installation** and **Operator Installation** sections of the [quickstart](../quickstart.md) guide.
```

File: quickstart.md [hunk 6]
```diff
@@ -99,6 +101,25 @@ Wait until each pod has the STATUS of Running.
...
+```
+kubectl apply -k github.com/confidential-containers/operator/config/samples/enclave-cc/sim?ref=<RELEASE_VERSION>
+```
+or
+```
+kubectl apply -k github.com/confidential-containers/operator/config/samples/enclave-cc/hw?ref=<RELEASE_VERSION>
+```
+for the **simulated** SGX mode build or **hardware** SGX mode build, respectively.
+
+**Note** For `enclave-cc` certain configuration changes, such as setting the
```
Comment (dcmiddle) [inline quickstart.md:117]: Delete this note if applied as suggested above.

**Review by mythi — COMMENTED** [review]
Just a quick comment on `enclave-cc` new image URLs. Do we need similar updates for the Kata CI images too?

**Comment by fitzthum** [comment]
@sameo @dcmiddle PTAL at updates

**Review by dcmiddle — APPROVED** [review]
Looks great. Thanks for the big reorg!

**Review by sameo — APPROVED** [review]

**Pull request merged**